src/gosling-track/gosling-track.ts

```diff
@@ -542,67 +542,84 @@ function GoslingTrack(HGC: import('@higlass/types').HGC, ...args: any[]): any {
+                if ('resolutions' in this.tilesetInfo) {
                     const sortedResolutions = this.tilesetInfo.resolutions
                         .map((x: number) => +x)
                         .sort((a: number, b: number) => b - a);
 
-                    this.xTiles = tileProxy.calculateTilesFromResolution(
-                        sortedResolutions[this.zoomLevel],
+                    const xTiles = tileProxy.calculateTilesFromResolution(
+                        sortedResolutions[zoomLevel],
                         this._xScale,
                         this.tilesetInfo.min_pos[0],
                         this.tilesetInfo.max_pos[0]
                     );
 
+                    let yTiles: number[] | undefined;
                     if (Is2DTrack(resolveSuperposedTracks(this.options.spec)[0])) {
                         // it makes sense only when the y-axis is being used for a genomic field
-                        this.yTiles = tileProxy.calculateTilesFromResolution(
-                            sortedResolutions[this.zoomLevel],
+                        yTiles = tileProxy.calculateTilesFromResolution(
+                            sortedResolutions[zoomLevel],
                             this._yScale,
                             this.tilesetInfo.min_pos[0],
                             this.tilesetInfo.max_pos[0]
                         );
                     }
 
-                    const tiles = this.tilesToId(this.xTiles, this.yTiles, this.zoomLevel);
+                    const tiles = this.tilesToId(xTiles, yTiles, zoomLevel);
+
                     this.setVisibleTiles(tiles);
                 } else {
-                    this.xTiles = tileProxy.calculateTiles(
-                        this.zoomLevel,
+                    const xTiles = tileProxy.calculateTiles(
+                        zoomLevel,
                         this.relevantScale(),
                         this.tilesetInfo.min_pos[0],
                         this.tilesetInfo.max_pos[0],
                         this.tilesetInfo.max_zoom,
                         this.tilesetInfo.max_width
                     );
 
+                    let yTiles: number[] | undefined;
                     if (Is2DTrack(resolveSuperposedTracks(this.options.spec)[0])) {
                         // it makes sense only when the y-axis is being used for a genomic field
-                        this.yTiles = tileProxy.calculateTiles(
-                            this.zoomLevel,
+                        yTiles = tileProxy.calculateTiles(
+                            zoomLevel,
                             this._yScale,
                             this.tilesetInfo.min_pos[1],
                             this.tilesetInfo.max_pos[1],
                             this.tilesetInfo.max_zoom,
-                            this.tilesetInfo.max_width1 || this.tilesetInfo.max_width
+                            // @ts-expect-error what is max_width1?
```

Comment:
yeah not sure about this!

Reply:
Hmm, not sure about it, too. I see it is used in HeatmapTiledPixiTrack.js of the higlass only.

Reply:
Ok I'll add it to that type...